File: Heat/Repositories/IHeatDBContext.vb
Imports System.Data.Entity
Imports Heat.Models

Public Interface IHeatDBContext
    Inherits IDisposable

    Property Addresses As DbSet(Of Address)
    Property AddressTypes As DbSet(Of AddressType)
    Property Customers As DbSet(Of Customer)
    Property Contacts As DbSet(Of Contact)
    Property Numberings As DbSet(Of Numbering)
    Property SerialSchemes As DbSet(Of SerialScheme)
    Property CausalWarehouses As DbSet(Of CausalWarehouse)
    Property CausalWarehouseGroups As DbSet(Of CausalWarehouseGroup)
    Property Warehouses As DbSet(Of Warehouse)
    Property WarehouseMovements As DbSet(Of WarehouseMovement)
    Property Sellers As DbSet(Of Seller)
    Property Invoices As DbSet(Of Invoice)
    Property InvoiceRows As DbSet(Of InvoiceRow)
    Property DocumentTypes As DbSet(Of DocumentType)
    Property Products As DbSet(Of Product)
    Property Payments As DbSet(Of Payment)
    Property ProductInvoiceRows As DbSet(Of ProductInvoiceRow)
    Property DescriptiveInvoiceRows As DbSet(Of DescriptiveInvoiceRow)
    Property ActionTypes As DbSet(Of ActionType)
    Property WorkOperators As DbSet(Of WorkOperator)
    Property WorkActions As DbSet(Of WorkAction)
    Property Operations As DbSet(Of Operation)
    Property Manifacturers As DbSet(Of Manifacturer)
    Property ManifacturerModels As DbSet(Of ManifacturerModel)
    
    Property HeatTransferFluids As DbSet(Of HeatTransferFluid)
    Property ThermalUnitKinds As DbSet(Of ThermalUnitKind)
    Property Plants As DbSet(Of Plant)
    Property PlantTypes As DbSet(Of PlantType)
    Property PlantClasses As DbSet(Of PlantClass)
    Property PlantServices As DbSet(Of PlantService)
    Property ThermalUnits As DbSet(Of ThermalUnit)
    Property Fuels As DbSet(Of Fuel)
    Property Heaters As DbSet(Of Heater)
    Property Media As DbSet(Of Medium)



    Function SaveChanges() As Integer

    Sub SetModified(entity As Object)

End Interface

File: Heat.ConvertedToC#/Repositories/IHeatDBContext.cs
using System;
using System.Data.Entity;
using Heat.Models;
namespace Heat
{

    public interface IHeatDBContext : IDisposable
	{

		IDbSet<Address> Addresses { get; set; }
		IDbSet<AddressType> AddressTypes { get; set; }
		DbSet<Customer> Customers { get; set; }
		IDbSet<Contact> Contacts { get; set; }
		DbSet<Numbering> Numberings { get; set; }
		IDbSet<SerialScheme> SerialSchemes { get; set; }
		IDbSet<CausalWarehouse> CausalWarehouses { get; set; }
		IDbSet<CausalWarehouseGroup> CausalWarehouseGroups { get; set; }
		IDbSet<Warehouse> Warehouses { get; set; }
		IDbSet<WarehouseMovement> WarehouseMovements { get; set; }
		IDbSet<Seller> Sellers { get; set; }
		IDbSet<Invoice> Invoices { get; set; }
		IDbSet<InvoiceRow> InvoiceRows { get; set; }
		IDbSet<DocumentType> DocumentTypes { get; set; }
		IDbSet<Product> Products { get; set; }
		IDbSet<Payment> Payments { get; set; }
		IDbSet<ProductInvoiceRow> ProductInvoiceRows { get; set; }
		IDbSet<DescriptiveInvoiceRow> DescriptiveInvoiceRows { get; set; }
		IDbSet<ActionType> ActionTypes { get; set; }
		IDbSet<WorkOperator> WorkOperators { get; set; }
		IDbSet<WorkAction> WorkActions { get; set; }
		IDbSet<Operation> Operations { get; set; }
		DbSet<Manifacturer> Manifacturers { get; set; }
		DbSet<ManifacturerModel> ManifacturerModels { get; set; }

		IDbSet<HeatTransferFluid> HeatTransferFluids { get; set; }
		IDbSet<ThermalUnitKind> ThermalUnitKinds { get; set; }
		DbSet<Plant> Plants { get; set; }
		DbSet<PlantType> PlantTypes { get; set; }
		DbSet<PlantClass> PlantClasses { get; set; }
		IDbSet<PlantService> PlantServices { get; set; }
		DbSet<ThermalUnit> ThermalUnits { get; set; }
		DbSet<Fuel> Fuels { get; set; }
		IDbSet<Heater> Heaters { get; set; }
		IDbSet<Medium> Media { get; set; }
		IDbSet<OutboundCall> OutboundCalls { get; set; }

		DbSet<ProposedOutBoundCall> ProposedOutboundCalls { get; set; }
        DbSet<AssignedOutboundCall> AssignedOutboundCalls { get; set; }
        DbSet<ProposedCallsGeneration> ProposedCallsGenerations { get; set; }

		int SaveChanges();
		void SetModified(object entity);
	}
}
